docs/intro/tutorial.rst

```diff
@@ -55,318 +55,333 @@ This will create a ``tutorial`` directory with the following contents::
+:class:`~scrapy.selector.SelectorList`, which represents a list of
+:class:`~scrapy.selector.Selector` objects that wrap around XML/HTML elements
+and allow you to run further queries to fine-grain the selection or extract the
+data.
 
-* :meth:`~scrapy.selector.Selector.css`: returns a list of selectors, each of
-  which represents the nodes selected by the CSS expression given as argument.
+To extract the text from the title above, you can do::
 
-* :meth:`~scrapy.selector.Selector.extract`: returns a unicode string with the
-  selected data.
+    >>> response.css('title::text').extract()
+    [u'Quotes to Scrape']
 
-* :meth:`~scrapy.selector.Selector.re`: returns a list of unicode strings
-  extracted by applying the regular expression given as argument.
+There are two things to note here: one is that we've added ``::text`` to the
+CSS query, to mean that we want to select the text from inside the title element.
+If we don't specify ``::text``, we'd get the HTML tags::
 
+    >>> response.css('title').extract()
+    [u'<title>Quotes to Scrape</title>']
 
-Trying Selectors in the Shell
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The other thing is that the result of calling ``.extract()`` is a list, because
+we're dealing with an instance of :class:`~scrapy.selector.SelectorList`.  When
+you know you just want the first result, as in this case, you can do::
 
-To illustrate the use of Selectors we're going to use the built-in :ref:`Scrapy
-shell <topics-shell>`, which also requires `IPython <http://ipython.org/>`_ (an extended Python console)
-installed on your system.
+    >>> response.css('title::text').extract_first()
+    u'Quotes to Scrape'
 
-To start a shell, you must go to the project's top level directory and run::
+As an alternative, you could've written::
 
-    scrapy shell "http://quotes.toscrape.com"
+    >>> response.css('title::text')[0].extract()
+    u'Quotes to Scrape'
 
-.. note::
+However, using ``.extract_first()`` avoids an ``IndexError`` and returns
+``None`` when it doesn't find any element matching the selection.
 
-   Remember to always enclose urls in quotes when running Scrapy shell from
-   command-line, otherwise urls containing arguments (ie. ``&`` character)
-   will not work.
+There's a lesson here: for most scraping code, you want it to be resilient to
+errors due to things not being found on a page, so that even if some parts fail
+to be scraped, you can at least get **some** data.
 
-This is what the shell looks like::
+Besides the :meth:`~scrapy.selector.Selector.extract` and
+:meth:`~scrapy.selector.SelectorList.extract_first` methods, you can also use
+the :meth:`~scrapy.selector.Selector.re` method to extract using a regular
+expression::
 
-    [ ... Scrapy log here ... ]
+    >>> response.css('title::text').re('Quotes.*')
+    [u'Quotes to Scrape']
+    >>> response.css('title::text').re('Q\w+')
+    [u'Quotes']
+    >>> response.css('title::text').re('(\w+) to (\w+)')
```

Comment:
I think it is a good practice to use raw strings for regexes (`r'(\w+) to (\w+)'`) because it allows to avoid gotchas with some regexes, e.g. with `\b`.